File: game-collection-frontend/src/app/model/game-details.ts
import {Platform} from "./platform";

export class GameDetails {
  slug: string;
  name: string;
  description: string;
  released: string;
  background_image: string;
  website: string;
  platforms: Platform[];
  genres: Genre[];
  tags: Tag[];
}

export class ScreenShot {
  id: number;
  image: string;
}

export class Genre {
  id: number;
  name: string;
}

export class Tag {
  id: number;
  name: string;
}

File: game-collection-backend/src/main/java/io/github/agawrysiuk/gamecollectionbackend/gamedetails/model/GameDetails.java
package io.github.agawrysiuk.gamecollectionbackend.gamedetails.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Set;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GameDetails {
    @Id
    private String slug;
    private String name;
    @Lob
    private String description;
    private String released;
    private String background_image;
    private String website;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JoinColumn(name = "platforms_id")
    private Set<Platform> platforms;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JoinColumn(name = "genre_id")
    private Set<Genre> genres;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JoinColumn(name = "tag_id")
    private Set<Tag> tags;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JoinColumn(name = "screenshot_id")
    private Set<ScreenShot> screenShots;
}
